weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/ObjectViewer.tsx

```diff
@@ -90,6 +124,22 @@ export const ObjectViewer = ({
     setExpandedRefs(eRefs => ({...eRefs, [path]: ref}));
   }, []);
 
+  // For each expanded id, if it is not in the `expandedRefs` state,
+  // We check if it resolves to a ref and if so we add it to the `expandedRefs` state.
+  // We need this, because expandAll adds a bunch of ids to expandedIds, but not to expandedRefs.
+  for (const id of expandedIds) {
+    if (typeof id === 'number') {
+      continue;
+    }
+    if (!(id in expandedRefs)) {
+      // ref is a ref or undefined
+      const ref = getRefFromObjectPath(resolvedData, id);
+      if (isWeaveRef(ref)) {
+        addExpandedRef(id, ref);
```

Comment:
This will create N (N=number of refs) rerenders and is subject to thrashing. let's refactor. See comment above